.github/workflows/notify-on-merge.js

```diff
@@ -0,0 +1,71 @@
+const { slackNotification, getLocalConfigs } = require('./helpers.js');
+
+const SLACK = {
+  merge: ({ html_url, number, title, prefix = '' }) =>
+    `:merged: PR merged to stage: ${prefix} <${html_url}|#${number}: ${title}>.`,
+};
+
+const getCommitSha = () => {
+  const commitSha = process.env.GITHUB_SHA;
+  if (!commitSha) {
+    throw new Error(
+      "GITHUB_SHA is not set. For local testing, please add a valid commit SHA in your .env file."
+    );
+  }
+  return commitSha;
+};
+
+const getMergedPRs = async (github, context) => {
+  const { owner, repo } = context.repo;
+  const commitSha = getCommitSha();
+  const { data: prs } = await github.rest.repos.listPullRequestsAssociatedWithCommit({
+    owner,
+    repo,
+    commit_sha: commitSha,
+  });
+  return prs.map(pr => ({
+    number: pr.number,
+    title: pr.title,
+    owner: pr.user.login,
+    html_url: pr.html_url,
+  }));
+};
+
+async function main({ github, context } = {}) {
+  if (!github || !context) {
+    if (process.env.LOCAL_RUN) {
+      console.log("Local run detected. Loading local configurations...");
+      const localConfigs = getLocalConfigs();
+      github = localConfigs.github;
+      context = localConfigs.context;
+    } else {
+      throw new Error("GitHub context is missing. Ensure you are running in the correct environment.");
+    }
+
+  }
+
+  try {
+    const prs = await getMergedPRs(github, context);
+    for (const pr of prs) {
+      const message = SLACK.merge({
+        html_url: pr.html_url,
+        number: pr.number,
+        title: pr.title,
+        prefix: '',
+      });
+      console.log(`Sending notification for PR #${pr.number}: ${pr.title}`);
+      await slackNotification(message, process.env.OKAN_SLACK_WEBHOOK);
+    }
+  } catch (error) {
+    console.error('Error fetching or notifying for PR(s):', error);
+  }
+
+
+}
+
+if (process.env.LOCAL_RUN) {
+  const { github, context } = getLocalConfigs();
+  main({ github, context });
+}
+
+module.exports = main;
```

Comment:
I think that you already have everything you need and got everything that you need. It's okay to fail the action on error, so we don't need to bloat the code with handling that would just hide that the message fails - so it's even counterproductive
```suggestion
const { slackNotification, getLocalConfigs } = require('./helpers.js');

async function main({ github, context } = {}) {
  if ((!github || !context) || process.env.LOCAL_RUN) throw new Error("...")
  if (process.env.LOCAL_RUN) {
    console.log('Local run detected. Loading local configurations...');
    const localConfigs = getLocalConfigs();
    github = localConfigs.github;
    context = localConfigs.context;
  } 

  const { pr: {
    number,
    title,
    html_url
  }} = context.payload;
  console.log(`Sending notification for PR #${number}: ${title}`);
  await slackNotification(`:merged: PR merged to stage <${html_url}|#${number}: ${title}>.`, process.env.SLACK_WEBHOOK);
}

if (process.env.LOCAL_RUN) {
  const { github, context } = getLocalConfigs();
  main({ github, context });
}

module.exports = main;
```